embedded/sql/engine.go

```diff
@@ -40,7 +40,10 @@ var ErrTableDoesNotExist = errors.New("table does not exist")
 var ErrColumnDoesNotExist = errors.New("column does not exist")
 var ErrColumnNotIndexed = errors.New("column is not indexed")
 var ErrLimitedKeyType = errors.New("indexed key of invalid type. Supported types are: INTEGER, VARCHAR[256] OR BLOB[256]")
+var ErrAutoIncrementWrongType = errors.New("auto incremented column need to be INTEGER type")
+var ErrAutoIncrementMultiple = errors.New("several auto incremental column were found. Wrong schema")
```

Comment:
given this is a generalization of the auto_increment attribute, it might be worth supporting multiple auto_increment columns... WDYT?

Reply:
technically it won't change anything, because the columns will all use use the same index.
but it's a bit useless because it will be a duplication of effort because the columns will always have the same value

Reply:
all auto_incremental columns will have the same value until we include support for altering the schema e.g. switching from auto_increment on and off... or setting a different initial value or delta.. but I think it's ok so have limitations along the way

Reply:
I propose to resolve this conversation for now. Then, when the feature will be available, open a new issue to support multi column auto_increment